pkg/tools/cli/loadtest/service/loader.go

```diff
@@ -138,44 +165,108 @@ func (l *loader) Do(ctx context.Context) <-chan error {
 		}
 		return nil
 	}))
-	start = time.Now()
-	var errCnt int32 = 0
+
 	l.eg.Go(safety.RecoverFunc(func() error {
+		log.Infof("start load test(%s, %s)", l.service.String(), l.operation.String())
 		defer close(ech)
 		eg, egctx := errgroup.New(ctx)
-		eg.Limitation(l.concurrency)
-		for _, req := range l.requests {
-			r := req
-			eg.Go(safety.RecoverFunc(
-				func() error {
+		switch l.operation {
+		case config.StreamInsert, config.StreamSearch:
+			var newData func() interface{}
+			switch l.operation {
+			case config.StreamInsert:
+				newData = func() interface{} {
+					return new(payload.Empty)
+				}
+			case config.StreamSearch:
+				newData = func() interface{} {
+					return new(payload.Search_Response)
+				}
+			}
+
+			start = time.Now()
+			eg.Go(safety.RecoverFunc(func() error {
+				// TODO: related to #557
+				/*
 					_, err := l.client.Do(egctx, l.addr, func(ctx context.Context, conn *grpc.ClientConn, copts ...grpc.CallOption) (interface{}, error) {
-						err := l.loaderFunc(ctx, vald.NewValdClient(conn), r, copts...)
-						atomic.AddInt32(&pgCnt, 1)
+						st, err := l.loaderFunc(ctx, conn, nil, copts...)
 						if err != nil {
-							log.Warn("an error occurred while executing loaderfunc:", err)
-							atomic.AddInt32(&errCnt, 1)
+							return nil, err
 						}
-						return nil, err
+						return nil, grpc.BidirectionalStreamClient(st.(grpc.ClientStream), l.dataProvider, newData, f)
 					})
+				*/
+				conn, err := grpc.Dial(l.addr, grpc.WithInsecure())
+				if err != nil {
+					finalize(err)
+					return nil
+				}
+				defer func() {
+					finalize(conn.Close())
+				}()
+				st, err := l.loaderFunc(egctx, conn, nil)
+				if err != nil {
+					finalize(err)
+					return nil
+				}
+				if err := igrpc.BidirectionalStreamClient(st.(grpc.ClientStream), l.dataProvider, newData, f); err != nil {
+					finalize(err)
+				}
+				return nil
+			}))
+			err = eg.Wait()
+			end = time.Now()
+		case config.Insert, config.Search:
+			eg.Limitation(l.concurrency)
+			start = time.Now()
+			for {
+				r := l.dataProvider()
+				if r == nil {
+					break
+				}
+				eg.Go(safety.RecoverFunc(func() error {
+					// TODO: related to #557
+					/*
+						_, err := l.client.Do(egctx, l.addr, func(ctx context.Context, conn *grpc.ClientConn, copts ...grpc.CallOption) (interface{}, error) {
+							res, err := l.loaderFunc(ctx, conn, r, copts...)
+							f(res, err)
+							return res, err
+						})
+					*/
+					conn, err := grpc.Dial(l.addr, grpc.WithInsecure())
```

Comment:
not sure if we should use the `WithInsecure` option

Reply:
This is just a temporary fix.
If it is enable to use vald internal client, I will fix it.